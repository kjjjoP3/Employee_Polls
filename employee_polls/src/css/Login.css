body {
    text-align: center;
    font-family: 'Arial', sans-serif;
  }
  
  .card {
    background: white;
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.1);
  }
  
  .alert {
    margin-bottom: 20px;
  }
  
  .form-label {
    font-weight: 500;
  }
  
  .form-control {
    border: 2px solid #ced4da;
    border-radius: 10px;
    transition: border-color 0.3s;
  }
  
  .form-control:focus {
    border-color: #007bff;
    box-shadow: 0 0 5px rgba(0, 123, 255, 0.5);
  }
  
  .btn {
    border: none;
    transition: background-color 0.3s;
  }
  
  .btn:hover {
    background-color: #0056b3;
  }
  
  .link-secondary {
    color: #007bff;
  }
  
  .link-secondary:hover {
    text-decoration: underline;
  }
  