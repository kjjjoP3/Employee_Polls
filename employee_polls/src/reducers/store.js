import { configureStore } from "@reduxjs/toolkit";
import authen from "./authenSlice";
import currentUser from "./currentUserSlice";
import questions from "./questionSlice";

const store = configureStore({
    reducer: {
        allUser: authen,
        currentUser: currentUser,
        questions: questions,
    },
});

export default store;
