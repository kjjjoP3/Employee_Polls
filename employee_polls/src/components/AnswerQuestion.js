import { useSelector, useDispatch } from "react-redux";
import { useNavigate, useParams } from "react-router-dom";
import { getQuestions, saveQuestionAnswer } from "../reducers/questionSlice";
import { useEffect, useState } from "react";
import { Card } from 'primereact/card';
import { Button } from 'primereact/button';
import { Divider } from 'primereact/divider';
import { Avatar } from 'primereact/avatar';

export const AnswerQuestion = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [isDisable, setIsDisable] = useState(false);
  const [isOption1, setIsOption1] = useState(false);
  const [isOption2, setIsOption2] = useState(false);
  const [voteOption1, setVoteOption1] = useState(0);
  const [voteOption2, setVoteOption2] = useState(0);
  const [currentUser, setCurrentUser] = useState([]);
  const [questions, setQuestions] = useState([]);
  const [users, setUsers] = useState([]);
  const [questionCurrent, setQuestionCurrent] = useState([]);
  const [authorCurrent, setAuthorCurrent] = useState([]); // Initialize authorCurrent as an array
  const a = useSelector((state) => state.currentUser.value);
  const b = useSelector((state) => state.questions.value);
  const c = useSelector((state) => state.allUser.value);
  const params = useParams();
  const questionId = params.question_id;

  const d = Object.values(b).filter((val) => val.id === questionId);

  useEffect(() => {
    dispatch(getQuestions());
  }, []);

  useEffect(() => {
    setCurrentUser(a);
    setQuestions(Object.values(b));
    setUsers(Object.values(c));

    if (!d.length) {
      navigate("/404");
    }

    setQuestionCurrent(d);
    const e = Object.values(c).filter((val) => val.id === d[0]?.author);
    setAuthorCurrent(e); // Set authorCurrent with the filtered users
    setIsOption1(d[0]?.optionOne.votes.includes(a[0]));
    setIsOption2(d[0]?.optionTwo.votes.includes(a[0]));
    setVoteOption1(d[0]?.optionOne.votes.length);
    setVoteOption2(d[0]?.optionTwo.votes.length);
    if (d[0]?.optionOne.votes.includes(a[0]) || d[0]?.optionTwo.votes.includes(a[0])) {
      setIsDisable(true);
    }
  }, [b]);

  const handleAnswer = (val) => {
    if (!isDisable) {
      setIsDisable(true);
      if (val === "optionOne") {
        setVoteOption1(voteOption1 + 1);
        setIsOption1(true);
      } else {
        setVoteOption2(voteOption2 + 1);
        setIsOption2(true);
      }
      const payload = { authedUser: currentUser[0], qid: questionId, answer: val };
      dispatch(saveQuestionAnswer(payload));
    }
  };

  const title = "Poll by " + questionCurrent[0]?.author;

  return (
    <div className="flex justify-content-center text-center">
      <div>
        <h2>{title}</h2>

        {authorCurrent[0]?.avatarURL && (
          <Avatar image={authorCurrent[0]?.avatarURL} size="large" shape="circle" />
        )}

        <h1>Would You Rather</h1>

        <div className="flex justify-content-center">
          <Card className="m-2" style={{ width: '400px' }}>
            <Button
              label={questionCurrent[0]?.optionOne?.text}
              className={isOption1 ? 'p-button-success' : 'p-button-outlined'}
              onClick={() => handleAnswer("optionOne")}
              disabled={isDisable}
            />
            {isDisable && (
              <div>
                <span>Voted people: {voteOption1}</span>
                <br />
                <span>Percentage: {((voteOption1 * 100) / (voteOption1 + voteOption2)).toFixed(2)}%</span>
              </div>
            )}
          </Card>
          <Divider layout="vertical" />
          <Card className="m-2" style={{ width: '400px' }}>
            <Button
              label={questionCurrent[0]?.optionTwo?.text}
              className={isOption2 ? 'p-button-success' : 'p-button-outlined'}
              onClick={() => handleAnswer("optionTwo")}
              disabled={isDisable}
            />
            {isDisable && (
              <div>
                <span>Voted people: {voteOption2}</span>
                <br />
                <span>Percentage: {((voteOption2 * 100) / (voteOption1 + voteOption2)).toFixed(2)}%</span>
              </div>
            )}
          </Card>
        </div>
      </div>
    </div>
  );
};
