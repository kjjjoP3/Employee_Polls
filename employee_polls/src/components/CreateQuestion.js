import { Button } from 'primereact/button';
import { InputText } from 'primereact/inputtext';
import { Card } from 'primereact/card';
import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { saveQuestion } from '../reducers/questionSlice';
import 'primeflex/primeflex.css';

export const CreateQuestion = () => {
  const [firstOption, setFirstOption] = useState('');
  const [secondOption, setSecondOption] = useState('');
  const [hasError, setHasError] = useState(false);

  const dispatch = useDispatch();
  const navigate = useNavigate();

  // Retrieve currentUser from session storage
  const storedUser = JSON.parse(sessionStorage.getItem('currentUser'));
  const currentUser = storedUser ? storedUser : [];

  const handleSubmit = () => {
    if (firstOption && secondOption) {
      const questionData = {
        optionOneText: firstOption,
        optionTwoText: secondOption,
        author: currentUser[0], // Assuming currentUser[0] contains the username
      };
      dispatch(saveQuestion(questionData));
      navigate('/home');
    } else {
      setHasError(true);
    }
  };

  return (
    <div className="flex justify-content-center align-items-center w-full h-full">
      {hasError && <h1 className="error">Please fill in both options</h1>}
      <Card>
        <h1>Would You Rather</h1>
        <h2>Create Your Own Poll</h2>
        <div className="options">
          <div>
            <h3>First Option</h3>
            <InputText 
              value={firstOption} 
              data-testid="optionOne" 
              onChange={(e) => setFirstOption(e.target.value)} 
            />
          </div>
          <div>
            <h3>Second Option</h3>
            <InputText 
              value={secondOption} 
              data-testid="optionTwo" 
              onChange={(e) => setSecondOption(e.target.value)} 
            />
          </div>
        </div>
        <div>
          <Button 
            label="Submit" 
            data-testid="add" 
            className="mt-2" 
            onClick={handleSubmit} 
          />
        </div>
      </Card>
    </div>
  );
};
