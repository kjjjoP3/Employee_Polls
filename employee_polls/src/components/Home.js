import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { TabView, TabPanel } from 'primereact/tabview';
import { DataView } from 'primereact/dataview'; 
import { Button } from 'primereact/button';
import { useNavigate, useLocation } from 'react-router-dom';
import { getQuestions } from "../reducers/questionSlice";  
import { Card } from 'primereact/card';
import { Message } from 'primereact/message';
import '../Home.css';

export const Home = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const loggedInUser = useSelector(state => state.currentUser.value);
  const questionsData = useSelector(state => state.questions.value);
  
  const dispatch = useDispatch();
  const [isNotificationVisible, setIsNotificationVisible] = useState(false);
  const [notificationContent, setNotificationContent] = useState("");

  useEffect(() => {
    dispatch(getQuestions());
  }, [dispatch]);

  useEffect(() => {
    if (!loggedInUser.length) {
      navigate('/');
    }
  }, [loggedInUser, navigate]);

  const incomingNotification = location.state?.success;

  useEffect(() => {
    if (incomingNotification) {
      setNotificationContent(incomingNotification);
      setIsNotificationVisible(true);
      const timer = setTimeout(() => {
        setIsNotificationVisible(false);
        setNotificationContent("");
      }, 2000);
      return () => clearTimeout(timer);
    }
  }, [incomingNotification]);

  const questionsArray = Object.values(questionsData);
  const sortedQuestions = JSON.parse(JSON.stringify(questionsArray));
  sortedQuestions.sort((firstItem, secondItem) => secondItem.timestamp - firstItem.timestamp);
  sortedQuestions.forEach(item => item.timestamp = new Date(item.timestamp).toLocaleString());

  const newQuestionsList = sortedQuestions.filter(item => 
    !item.optionOne.votes.includes(loggedInUser[0]) &&
    !item.optionTwo.votes.includes(loggedInUser[0])
  );

  const completedQuestionsList = sortedQuestions.filter(item => 
    item.optionOne.votes.includes(loggedInUser[0]) ||
    item.optionTwo.votes.includes(loggedInUser[0])
  );

  const renderQuestionItem = (item) => {
    return (
      <div className="col-12 md:col-4">
        <Card className="m-2">
          <div className="flex flex-column p-3">
            <div className="flex flex-column justify-content-between align-items-start flex-1">
              <div className="flex flex-column align-items-start gap-2">
                <div className="text-xl font-bold">{item.author}</div>      
                <div className="text-gray-600">{item.timestamp}</div>      
              </div>
              <div className="flex align-items-center">
                <Button label="Show" onClick={() => navigate(`/questions/${item.id}`)} />
              </div>
            </div>
          </div>
        </Card>
      </div>
    );
  };

  return (
    <div className="p-4">
      {isNotificationVisible && (
        <Message severity="info" text={notificationContent} />
      )}
      <TabView>
        <TabPanel header="New Questions">
          <DataView value={newQuestionsList} itemTemplate={renderQuestionItem} layout="grid" />
        </TabPanel>
        <TabPanel header="Done">
          <DataView value={completedQuestionsList} itemTemplate={renderQuestionItem} layout="grid" />
        </TabPanel>
      </TabView>
    </div>
  );
};
