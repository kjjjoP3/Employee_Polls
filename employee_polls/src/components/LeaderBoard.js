import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getQuestions } from "../reducers/questionSlice";
import { DataTable } from "primereact/datatable";
import { Column } from "primereact/column";
import { Card } from "primereact/card";
import { useNavigate } from "react-router-dom";

export const LeaderBoard = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const users = useSelector((state) => state.allUser.value);
  const questions = useSelector((state) => state.questions.value);

  const leaderboardData = [];

  useEffect(() => {
    dispatch(getQuestions());
  }, [dispatch]);

  useEffect(() => {
    const currentUser = JSON.parse(sessionStorage.getItem("currentUser"));
    if (!currentUser || currentUser.length === 0) {
      navigate("/");
    }
  }, [navigate]);

  Object.values(users).forEach((user) => {
    const answeredCount = Object.values(questions).filter(
      (question) =>
        question.optionOne.votes.includes(user.id) ||
        question.optionTwo.votes.includes(user.id)
    ).length;

    const createdCount = Object.values(questions).filter(
      (question) => question.author === user.id
    ).length;

    leaderboardData.push({
      name: user.name,
      avatarURL: user.avatarURL,
      answered: answeredCount,
      created: createdCount,
      score: answeredCount + createdCount,
    });
  });

  if (
    (users && Object.keys(users).length > 0) ||
    (questions && Object.keys(questions).length > 0)
  ) {
    sessionStorage.setItem("leaderboardData", JSON.stringify(leaderboardData));
  }

  let leaderboardSession =
    JSON.parse(sessionStorage.getItem("leaderboardData")) || [];

  if (leaderboardSession.length > 0) {
    leaderboardSession.sort((a, b) => b.score - a.score);
  }

  return (
    <div className="p-4">
      <Card title="Leader Board" className="shadow-2">
        <DataTable value={leaderboardData} paginator rows={10} className="mt-3">
          <Column
            header="Users"
            body={(rowData) => (
              <div className="flex align-items-center">
                {rowData.avatarURL ? (
                  <img
                    src={rowData.avatarURL}
                    alt={rowData.name}
                    height="40"
                    className="mr-2"
                  />
                ) : (
                  <div
                    className="mr-2"
                    style={{
                      height: "40px",
                      width: "40px",
                      backgroundColor: "#f0f0f0",
                      borderRadius: "50%",
                    }}
                  ></div>
                )}
                <span>{rowData.name}</span>
              </div>
            )}
          />
          <Column field="answered" header="Answered" />
          <Column field="created" header="Created" />
        </DataTable>
      </Card>
    </div>
  );
};
