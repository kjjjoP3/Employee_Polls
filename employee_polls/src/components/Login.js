import { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useNavigate, useLocation } from "react-router-dom";
import { getUsers } from "../reducers/authenSlice";
import { currentUser } from "../reducers/currentUserSlice";
import { Card } from 'primereact/card';
import { InputText } from 'primereact/inputtext';
import { Button } from 'primereact/button';
import { Message } from 'primereact/message';

export const Login = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const [userInput, setUserInput] = useState({ username: "", password: "" });
  const [errorVisible, setErrorVisible] = useState(false);
  const usersList = useSelector((state) => state.allUser.value); // Ensure you access the 'value' property
  const loggedInUser = useSelector((state) => state.currentUser.value);

  useEffect(() => {
    if (loggedInUser.length > 0) {
      navigate('/home', { replace: true }); // Redirect logged-in users to home
    } else {
      dispatch(getUsers());
    }
  }, [loggedInUser, dispatch, navigate]);

  const handleLogin = () => {
    const { username, password } = userInput;

    if (username && password) {
      const foundUser = Object.entries(usersList).find(
        ([, user]) => user.id === username && user.password === password
      );

      if (foundUser) {
        dispatch(currentUser(foundUser));
        setErrorVisible(false);
        setUserInput({ username: "", password: "" });
        navigate('/home', { replace: true, state: { success: "Login Successful!" } });
      } else {
        setErrorVisible(true);
      }
    } else {
      setErrorVisible(true);
    }
  };

  return (
    <div className="flex justify-content-center align-items-center vh-100">
      <Card title="Employee Polls" className="shadow" style={{ width: '400px' }}>
        <h3 className="text-center mb-4">Login</h3>
        {errorVisible && (
          <Message severity="error" text="Incorrect Username or Password" />
        )}
        <form>
          <div className="mb-4">
            <label htmlFor="username" className="p-float-label">
              <InputText
                data-testid="username-input"
                id="username"
                value={userInput.username}
                onChange={(e) => setUserInput({ ...userInput, username: e.target.value })}
                required
              />
              <label htmlFor="username">Username</label>
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="password" className="p-float-label">
              <InputText
                data-testid="password-input"
                id="password"
                type="password"
                value={userInput.password}
                onChange={(e) => setUserInput({ ...userInput, password: e.target.value })}
                required
              />
              <label htmlFor="password">Password</label>
            </label>
          </div>
          <Button
            type="button"
            label="Login"
            className="w-full"
            onClick={handleLogin}
            data-testid="submit"
          />
        </form>
      </Card>
    </div>
  );
};

export default Login;
