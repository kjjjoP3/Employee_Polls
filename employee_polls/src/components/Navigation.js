import { useEffect } from "react"; // Import useEffect here
import { Menubar } from "primereact/menubar";
import { Button } from "primereact/button";
import { useDispatch } from "react-redux";
import { useNavigate, useLocation } from "react-router-dom";
import { logout } from "../reducers/currentUserSlice";
import logo from "./../logo.svg";

const Navigation = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation(); // Get location for checking the current path

  // Retrieve user data from session storage
  const storedUser = JSON.parse(sessionStorage.getItem('currentUser'));
  const userData = storedUser ? storedUser : []; // If not found, set to empty array

  const handleLogout = () => {
    dispatch(logout());
    sessionStorage.removeItem('currentUser'); // Clear session storage on logout
    navigate("/");
  };

  const menuItems = [
    { label: "Home", icon: "pi pi-home", command: () => navigate("/home") },
    {
      label: "Leaderboard",
      icon: "pi pi-chart-bar",
      command: () => navigate("/leaderboard"),
    },
    {
      label: "New Poll",
      icon: "pi pi-plus-circle",
      command: () => navigate("/add"),
    },
  ];

  const logoElement = (
    <img alt="App Logo" src={logo} height="40" className="mr-2" />
  );

  const userElement = (
    <div className="flex align-items-center">
      {userData[1]?.avatarURL && (
        <img
          src={userData[1].avatarURL}
          alt="User Avatar"
          height="40"
          className="mr-2 border-circle"
        />
      )}
      <span className="mr-4">{userData[0]}</span>
      <Button
        label="Logout"
        icon="pi pi-sign-out"
        text
        onClick={handleLogout}
      />
    </div>
  );

  // Redirect to home if user is already logged in and tries to access login page
  useEffect(() => {
    const checkLoginRedirect = () => {
      if (userData.length > 0 && location.pathname === "/") {
        navigate("/home", { replace: true });
      }
    };
    checkLoginRedirect();
  }, [userData, navigate, location.pathname]);

  return (
    <div className="card">
      <Menubar model={menuItems} start={logoElement} end={userElement} />
    </div>
  );
};

export default Navigation;
