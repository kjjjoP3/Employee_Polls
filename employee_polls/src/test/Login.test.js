import '@testing-library/jest-dom';  // Add this import
import { fireEvent, render } from "@testing-library/react";
import { Login } from "../components/Login";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../reducers/store";
import * as React from "react";

describe("Login page", () => {
  it("will match to snapshot", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("will show an error message if the username or password is incorrect", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );

    const username = component.getByTestId("username-input");
    const password = component.getByTestId("password-input");
    const submit = component.getByTestId("submit");

    fireEvent.change(username, { target: { value: "wrongUser" } });
    fireEvent.change(password, { target: { value: "wrongPass" } });
    fireEvent.click(submit);

    // Check that the error message is displayed
    expect(component.getByText("Incorrect Username or Password")).toBeInTheDocument();
  });
});
