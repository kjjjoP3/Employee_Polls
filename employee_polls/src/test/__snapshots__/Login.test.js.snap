// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page will match to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-content-center align-items-center vh-100"
      >
        <div
          class="shadow p-card p-component"
          data-pc-name="card"
          data-pc-section="root"
          style="width: 400px;"
        >
          <div
            class="p-card-body"
            data-pc-section="body"
          >
            <div
              class="p-card-title"
              data-pc-section="title"
            >
              Employee Polls
            </div>
            <div
              class="p-card-content"
              data-pc-section="content"
            >
              <h3
                class="text-center mb-4"
              >
                Login
              </h3>
              <form>
                <div
                  class="mb-4"
                >
                  <label
                    class="p-float-label"
                    for="username"
                  >
                    <input
                      class="p-inputtext p-component"
                      data-pc-name="inputtext"
                      data-pc-section="root"
                      data-testid="username-input"
                      id="username"
                      required=""
                      value=""
                    />
                    <label
                      for="username"
                    >
                      Username
                    </label>
                  </label>
                </div>
                <div
                  class="mb-4"
                >
                  <label
                    class="p-float-label"
                    for="password"
                  >
                    <input
                      class="p-inputtext p-component"
                      data-pc-name="inputtext"
                      data-pc-section="root"
                      data-testid="password-input"
                      id="password"
                      required=""
                      type="password"
                      value=""
                    />
                    <label
                      for="password"
                    >
                      Password
                    </label>
                  </label>
                </div>
                <button
                  aria-label="Login"
                  class="w-full p-button p-component"
                  data-pc-name="button"
                  data-pc-section="root"
                  data-testid="submit"
                  type="button"
                >
                  <span
                    class="p-button-label p-c"
                    data-pc-section="label"
                  >
                    Login
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-content-center align-items-center vh-100"
    >
      <div
        class="shadow p-card p-component"
        data-pc-name="card"
        data-pc-section="root"
        style="width: 400px;"
      >
        <div
          class="p-card-body"
          data-pc-section="body"
        >
          <div
            class="p-card-title"
            data-pc-section="title"
          >
            Employee Polls
          </div>
          <div
            class="p-card-content"
            data-pc-section="content"
          >
            <h3
              class="text-center mb-4"
            >
              Login
            </h3>
            <form>
              <div
                class="mb-4"
              >
                <label
                  class="p-float-label"
                  for="username"
                >
                  <input
                    class="p-inputtext p-component"
                    data-pc-name="inputtext"
                    data-pc-section="root"
                    data-testid="username-input"
                    id="username"
                    required=""
                    value=""
                  />
                  <label
                    for="username"
                  >
                    Username
                  </label>
                </label>
              </div>
              <div
                class="mb-4"
              >
                <label
                  class="p-float-label"
                  for="password"
                >
                  <input
                    class="p-inputtext p-component"
                    data-pc-name="inputtext"
                    data-pc-section="root"
                    data-testid="password-input"
                    id="password"
                    required=""
                    type="password"
                    value=""
                  />
                  <label
                    for="password"
                  >
                    Password
                  </label>
                </label>
              </div>
              <button
                aria-label="Login"
                class="w-full p-button p-component"
                data-pc-name="button"
                data-pc-section="root"
                data-testid="submit"
                type="button"
              >
                <span
                  class="p-button-label p-c"
                  data-pc-section="label"
                >
                  Login
                </span>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
